Sources/MongoSwift/BSON/BSONEncoder.swift

```diff
@@ -47,12 +47,32 @@ public class BSONEncoder {
         case binary
     }
 
+    /// Enum representing the various strategies for encoding `Data`s.
+    ///
+    /// As per the BSON specification, the default strategy is to encode `Data`s as BSON binary types with the generic
+    /// binary subtype.
+    ///
+    /// - SeeAlso: bsonspec.org
+    public enum DataEncodingStrategy {
+        /// Encode the `Data` by deferring to its default encoding implementation.
+        ///
+        /// Note: The default encoding implementation attempts to encode the `Data` as a [UInt8], but because BSON does
+        /// not support integer types besides `Int32` or `Int64`, it actually gets encoded to BSON as an `[Int]`.
```

Comment:
```suggestion
        /// not support integer types besides `Int32` or `Int64`, it actually gets encoded to BSON as an `[Int32]`.
```
nit, it's really a 32-bit int - we just return the value as an `Int` when you retrieve it from a doc since those are more widely used in BSON.

maybe worth also adding a sentence explaining why that is undesirable (space inefficient).